main.go

```diff
@@ -175,9 +182,33 @@ func main() {
 		os.Exit(1)
 	}
 
-	setupLog.Info("starting manager")
-	if err := mgr.Start(ctx); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+	if !dryRun {
+		setupLog.Info("starting manager")
+		if err := mgr.Start(ctx); err != nil {
+			setupLog.Error(err, "problem running manager")
+			os.Exit(1)
+		}
+	} else {
+
+		setupLog.Info("Starting Plan for controller class " + controllerClass)
+		controlPlanes := &controlplanev1alpha1.KopsControlPlaneList{}
+		err := mgr.GetAPIReader().List(ctx, controlPlanes)
+		if err != nil {
+			setupLog.Error(err, "Error listing Kops Control Planes")
+			os.Exit(1)
+		}
+
+		for _, kcp := range controlPlanes.Items {
+			if kcp.Spec.ControllerClass == controllerClass {
+				setupLog.Info("Starting plan for Kops Control Plane " + kcp.Name)
+				_, err := controller.Reconcile(context.WithValue(context.WithValue(ctx, controlplane.ClientKey{}, mgr.GetAPIReader()), controlplane.KCPKey{}, kcp), ctrl.Request{})
```

Comment:
I think this will be easier to understand by populating the context in other lines and just passing here